esp-hal-common/src/rtc_cntl/rtc/esp32c6.rs

```diff
@@ -0,0 +1,697 @@
+                unsafe {
+                    ets_delay_us(5);
+                }
+            }
+        }
+
+        if cal_clk == RtcCalSel::RtcCal32kRc {
+            if rc32k_enabled {
+                pmu.hp_sleep_lp_ck_power
+                    .modify(|_, w| w.hp_sleep_xpd_rc32k().set_bit());
+                unsafe {
+                    ets_delay_us(300);
+                }
+            }
+            if dig_rc32k_enabled {
+                lp_clkrst
+                    .clk_to_hp
+                    .modify(|_, w| w.icg_hp_osc32k().set_bit());
+            }
+        }
+
+        cal_val
+    }
+
+    /// Measure ratio between XTAL frequency and RTC slow clock frequency
+    fn get_calibration_value(cal_clk: RtcCalSel, slowclk_cycles: u32) -> u32 {
+        let xtal_freq = RtcClock::get_xtal_freq();
+        let xtal_cycles = RtcClock::calibrate_internal(cal_clk, slowclk_cycles) as u64;
+        let divider = xtal_freq.mhz() as u64 * slowclk_cycles as u64;
+        let period_64 = ((xtal_cycles << RtcClock::CAL_FRACT) + divider / 2u64 - 1u64) / divider;
+
+        (period_64 & u32::MAX as u64) as u32
+    }
+
+    /// Measure RTC slow clock's period, based on main XTAL frequency
+    ///
+    /// This function will time out and return 0 if the time for the given
+    /// number of cycles to be counted exceeds the expected time twice. This
+    /// may happen if 32k XTAL is being calibrated, but the oscillator has
+    /// not started up (due to incorrect loading capacitance, board design
+    /// issue, or lack of 32 XTAL on board).
+    fn calibrate(cal_clk: RtcCalSel, slowclk_cycles: u32) -> u32 {
+        let xtal_freq = RtcClock::get_xtal_freq();
+        let xtal_cycles = RtcClock::calibrate_internal(cal_clk, slowclk_cycles) as u64;
+        let divider = xtal_freq.mhz() as u64 * slowclk_cycles as u64;
+        let period_64 = ((xtal_cycles << RtcClock::CAL_FRACT) + divider / 2u64 - 1u64) / divider;
+
+        (period_64 & u32::MAX as u64) as u32
+    }
+
+    /// Calculate the necessary RTC_SLOW_CLK cycles to complete 1 millisecond.
+    pub(crate) fn cycles_to_1ms() -> u16 {
+        let period_13q19 = RtcClock::calibrate(
+            match RtcClock::get_slow_freq() {
+                RtcSlowClock::RtcSlowClockRcSlow => RtcCalSel::RtcCalRtcMux,
+                RtcSlowClock::RtcSlowClock32kXtal => RtcCalSel::RtcCal32kXtal,
+                RtcSlowClock::RtcSlowClock32kRc => RtcCalSel::RtcCal32kRc,
+                RtcSlowClock::RtcSlowOscSlow => RtcCalSel::RtcCal32kOscSlow,
+                // RtcSlowClock::RtcCalRcFast => RtcCalSel::RtcCalRcFast,
+            },
+            1024,
+        );
+
+        // 100_000_000 is used to get rid of `float` calculations
+        let period = (100_000_000 * period_13q19 as u64) / (1 << RtcClock::CAL_FRACT);
+
+        (100_000_000 * 1000 / period) as u16
+    }
+
+    pub(crate) fn estimate_xtal_frequency() -> u32 {
```

Comment:
Should this function be called in the other stubbed out estimate_xtal_frequency function?

Reply:
This should be removed altogether if we're removing the function in `rtc_cntl/mod.rs`, no?